### PR TITLE
Fix refpurpose of Imagick

### PR DESCRIPTION
**Pull request by LoginovIlya**
**Changes:**
- Fix refpurpose of Imagick
- Update description and parameters: reference/imagick/imagickkernel/scale.xml
- Fix type and add description: reference/imagick/imagick/setimagebiasquantum.xml
- Add description: reference/imagick/imagick/setimageattribute.xml

**Doc-base result:**
All good. Saving .manual.xml... done.

### CODE REVIEW

**Comment by Girgias** [comment]
@Danack can you have a look?

File: reference/imagick/imagick/getquantum.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 <refentry xml:id="imagick.getquantum" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
  <refnamediv>
   <refname>Imagick::getQuantum</refname>
-  <refpurpose>Description</refpurpose>
+  <refpurpose>Returns the ImageMagick quantum range as an integer</refpurpose>
```
Comment (Danack) [inline reference/imagick/imagick/getquantum.xml:7]: This is a bit complicated. The type for "quantum range" depends on whether ImageMagick was compiled with support for High Dynamic Range Images or not. If HDRI was enabled then the type is a float, if not it is an int.

So probably for this bit, just leave off the "as an integer".

File: reference/imagick/imagick/setimagebiasquantum.xml [hunk 2]
```diff
@@ -15,14 +15,12 @@
   &reftitle.description;
   <methodsynopsis>
    <modifier>public</modifier> <type>void</type><methodname>Imagick::setImageBiasQuantum</methodname>
-   <methodparam><type>string</type><parameter>bias</parameter></methodparam>
+   <methodparam><type>float</type><parameter>bias</parameter></methodparam>
   </methodsynopsis>
```
Comment (Danack) [inline reference/imagick/imagick/setimagebiasquantum.xml:19]: It should be a float....but it looks like currently it is a string. I've opened a bug: https://github.com/Imagick/imagick/issues/569

File: reference/imagick/imagick/statisticimage.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
 <refentry xml:id="imagick.statisticimage" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
  <refnamediv>
   <refname>Imagick::statisticImage</refname>
-  <refpurpose>Description</refpurpose>
+  <refpurpose>Replace each pixel with corresponding statistic from the neighborhood of the specified width and height.</refpurpose>
```
Comment (Danack) [inline reference/imagick/imagick/statisticimage.xml:7]: Maybe - `Replace each pixel with calculated value from the chosen statistic method of the specified width and height.` ?

**Review by Danack — COMMENTED** [review]
Hey, thanks for the PR.

All looks good apart from 2 simple things and 1 awkward one.

File: reference/imagick/imagick/statisticimage.xml (hunk 3 above)
Comment (cmb69) [reply reference/imagick/imagick/statisticimage.xml:7]: Is it possible to shorten this? The refpurpose is displayed on the [overview page](https://www.php.net/imagick) and long texts are not nice to read there.

File: reference/imagick/imagick/setimagebiasquantum.xml (hunk 2 above)
Comment (LoginovIlya) [reply reference/imagick/imagick/setimagebiasquantum.xml:19]: Got it. So what are we gonna do next? Update documentation now or wait until the bug will be fixed?

Comment (LoginovIlya) [reply reference/imagick/imagick/setimagebiasquantum.xml:19]: Then i update this based on this comments:
https://github.com/Imagick/imagick/blob/661405abe21d12003207bc8eb0963fafc2c02ee4/imagick_class.c#L5551

File: reference/imagick/imagick/statisticimage.xml (hunk 3 above)
Comment (LoginovIlya) [reply reference/imagick/imagick/statisticimage.xml:7]: I think this description cant be smaller. Like: ImagickDraw::pushPattern
If anything. I wait your decision @Danack , @cmb69.

File: reference/imagick/imagick/getquantum.xml (hunk 1 above)
Comment (LoginovIlya) [reply reference/imagick/imagick/getquantum.xml:7]: +

File: reference/imagick/imagick/statisticimage.xml (hunk 3 above)
Comment (Danack) [reply reference/imagick/imagick/statisticimage.xml:7]: How about `Modifies image using a statistics function` ?

File: reference/imagick/imagick/setimagebiasquantum.xml (hunk 2 above)
Comment (LoginovIlya) [reply reference/imagick/imagick/setimagebiasquantum.xml:19]: What about that? @Danack

Comment (LoginovIlya) [reply reference/imagick/imagick/setimagebiasquantum.xml:19]: What about that? @Danack

**Comment by LoginovIlya** [comment]
I fixed all

**Comment by LoginovIlya** [comment]
Up
@Danack @cmb69 @Girgias

File: reference/imagick/imagick/getregistry.xml [hunk 4]
```diff
@@ -4,7 +4,7 @@
 <refentry xml:id="imagick.getregistry" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
  <refnamediv>
   <refname>Imagick::getRegistry</refname>
-  <refpurpose>Description</refpurpose>
+  <refpurpose>Get the StringRegistry entry for the named key or false if not set</refpurpose>
```
Comment (cmb69) [inline reference/imagick/imagick/getregistry.xml:7]: This looks overly specific.
```suggestion
  <refpurpose>Get the StringRegistry entry for the named key</refpurpose>
```

File: reference/imagick/imagick/identifyformat.xml [hunk 5]
```diff
@@ -4,7 +4,7 @@
 <refentry xml:id="imagick.identifyformat" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
  <refnamediv>
   <refname>Imagick::identifyFormat</refname>
-  <refpurpose>Description</refpurpose>
+  <refpurpose>Replaces any embedded formatting characters with the appropriate image property and returns the interpreted text</refpurpose>
```
Comment (cmb69) [inline reference/imagick/imagick/identifyformat.xml:7]: This refpurpose appears to be way too long. I'm not sure what that method is supposed to do, though.

File: reference/imagick/imagick/setimageattribute.xml [hunk 6]
```diff
@@ -4,7 +4,7 @@
 <refentry xml:id="imagick.setimageattribute" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
  <refnamediv>
   <refname>Imagick::setImageAttribute</refname>
-  <refpurpose>Description</refpurpose>
+  <refpurpose>Sets the image attribute</refpurpose>
```
Comment (cmb69) [inline reference/imagick/imagick/setimageattribute.xml:7]:
```suggestion
  <refpurpose>Sets an image attribute</refpurpose>
```

File: reference/imagick/imagick/setimageattribute.xml [hunk 7]
```diff
@@ -19,11 +19,9 @@
    <methodparam><type>string</type><parameter>value</parameter></methodparam>
   </methodsynopsis>
   <para>
-
+   Sets the image attribute
```
Comment (cmb69) [inline reference/imagick/imagick/setimageattribute.xml:22]:
```suggestion
   Sets an image attribute.
```

File: reference/imagick/imagick/setprogressmonitor.xml [hunk 8]
```diff
@@ -4,7 +4,7 @@
 <refentry xml:id="imagick.setprogressmonitor" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
  <refnamediv>
   <refname>Imagick::setProgressMonitor</refname>
-  <refpurpose>Description</refpurpose>
+  <refpurpose>Set a callback that will be called during the processing of the Imagick image.</refpurpose>
```
Comment (cmb69) [inline reference/imagick/imagick/setprogressmonitor.xml:7]: This is a bit verbose. Maybe shorten:
```suggestion
  <refpurpose>Set a callback that will be called during the processing of the image</refpurpose>
```

File: reference/imagick/imagick/setimagebiasquantum.xml [hunk 9]
```diff
@@ -15,14 +15,12 @@
...
   <para>
-
+   Sets the image bias. Bias should be scaled with 0 = no adjustment, 1 = quantum value.
```
Comment (cmb69) [inline reference/imagick/imagick/setimagebiasquantum.xml:21]:
```suggestion
   Sets the image bias. Bias should be scaled with <literal>0</literal> (no adjustment) to <literal>1</literal> (quantum value).
```

File: reference/imagick/imagickkernel/addkernel.xml [hunk 10]
```diff
@@ -4,7 +4,7 @@
 <refentry xml:id="imagickkernel.addkernel" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
  <refnamediv>
   <refname>ImagickKernel::addKernel</refname>
-  <refpurpose>Description</refpurpose>
+  <refpurpose>Attach another kernel to this kernel to allow them to both be applied in a single morphology or filter function</refpurpose>
```
Comment (cmb69) [inline reference/imagick/imagickkernel/addkernel.xml:7]: I think this should be shortened as well. I have no idea how, though.

File: reference/imagick/imagickkernel/addunitykernel.xml [hunk 11]
```diff
@@ -4,7 +4,7 @@
 <refentry xml:id="imagickkernel.addunitykernel" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
  <refnamediv>
   <refname>ImagickKernel::addUnityKernel</refname>
-  <refpurpose>Description</refpurpose>
+  <refpurpose>Adds a given amount of the 'Unity' Convolution Kernel to the given pre-scaled and normalized Kernel</refpurpose>
```
Comment (cmb69) [inline reference/imagick/imagickkernel/addunitykernel.xml:7]: I think this should be shortened as well. I have no idea how, though.

File: reference/imagick/imagickkernel/scale.xml [hunk 12]
```diff
@@ -15,22 +15,41 @@
...
+     </listitem>
+    </varlistentry>
+    <varlistentry>
+     <term><parameter>normalizeFlag</parameter></term>
+     <listitem>
+      <para>
+       Imagick::NORMALIZE_KERNEL_NONE = 0
+       Imagick::NORMALIZE_KERNEL_VALUE = 8192
+       Imagick::NORMALIZE_KERNEL_CORRELATE = 65536
+       Imagick::NORMALIZE_KERNEL_PERCENT = 4096
```
[the remark this answers is not in the document]
Comment (Danack) [reply reference/imagick/imagickkernel/scale.xml:47]: Actually, exposing the constant values is probably not a great idea, as they aren't necessarily guaranteed across versions.

**Comment by Danack** [comment]
I rebased and made some small edits on #2149

**Comment by Danack** [comment]
Thanks. I committed these as 1ef9c7a76700b3e72844050d75e6ed1b870f9ca7 and  85278ca11d7a1e009af96b74448523e82ad99940 after rebasing.

**Pull request closed without merging**